lib/cgi/util.rb

```diff
@@ -10,19 +10,28 @@ module CGI::Util
   #   url_encoded_string = CGI::escape("'Stop!' said Fred")
   #      # => "%27Stop%21%27+said+Fred"
   def escape(string)
-    encoding = string.encoding
-    string.b.gsub(/([^ a-zA-Z0-9_.\-~]+)/) do |m|
-      '%' + m.unpack('H2' * m.bytesize).join('%').upcase
-    end.tr(' ', '+').force_encoding(encoding)
+    str = string.b
+    str.gsub!(/([^ a-zA-Z0-9_.\-~]+)/) do |m|
```

Comment:
I believe you can replace ``a-zA-Z0-9_`` with ``\w``.

Reply:
I'm not very good with regexp so if somebody else can confirm it.
It doesn't change performance.